test-e2e/utils/payments.ts

```diff
@@ -129,3 +130,24 @@ export function overrideIP(I: CodeceptJS.I) {
   // Set this as a cookie so it persists between page navigations (local storage would also work, but the permissions don't work)
   I.setCookie({ name: overrideIPCookieKey, value: '5.132.0.0', domain: 'localhost', path: '/' });
 }
+
+export async function registerAndSubscribe(I: CodeceptJS.I, props: ProviderProps, today: Date) {
+  const user = await I.registerOrLogin();
+
+  await goToCheckout(I);
+
+  I.waitForElement('#card', longTimeout);
+
+  I.payWithCreditCard(
```

Comment:
It looks like there's a missing await here (and also in the code you moved it from in `I can finish my subscription with credit card`.) That's probably why the tests were failing in the first place and refactoring here probably just happens to luckily tweak the operations so that it masks the race condition.

Do you use an IDE? I realize that this is not your code, but an IDE should flag things like this to help you when you are debugging. I highly recommend jetbrains products. Either way, I will look to improve our code inspection rules to prevent code being committed that has oversights like this.

Reply:
Same as the other comment, I believe that it's not necessary to await I.* calls unless the return value from them is needed

Reply:
Also worth noting that these are custom `I` extension methods, so you really have to look at the internals of them before assuming something from the docs applies.